### PR TITLE
genericapiserver: cut off pkg/serviceaccount dependency

### PR DESCRIPTION
**Pull request by sttts**
**Blocked** by pkg/api/validation/genericvalidation to be split up and moved into apimachinery.

### CODE REVIEW

**Comment by deads2k** [comment]
lgtm, squash bazel.

File: pkg/api/validation/genericvalidation/objectmeta.go [hunk 1]
```diff
@@ -44,67 +44,8 @@ var BannedOwners = map[schema.GroupVersionKind]struct{}{
...
-// trailing dashes are allowed.
-var ValidateNamespaceName = NameIsDNSLabel
-
 // ValidateClusterName can be used to check whether the given cluster name is valid.
-var ValidateClusterName = NameIsDNS1035Label
-
-// ValidateServiceAccountName can be used to check whether the given service account name is valid.
-// Prefix indicates this name will be used as part of generation, in which case
-// trailing dashes are allowed.
-var ValidateServiceAccountName = NameIsDNSSubdomain
```
Comment (deads2k) [inline pkg/api/validation/genericvalidation/objectmeta.go]: This one isn't an obvious move to me.

File: staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 	"fmt"
 	"strings"
 
-	"k8s.io/kubernetes/pkg/api/validation/genericvalidation"
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go]: uh oh.  We were able to merge this?  it should have failed a cyclical verify test.

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go]: No, your cyclic import script worked perfectly!

File: pkg/api/validation/genericvalidation/objectmeta.go (hunk 1 above)
Comment (sttts) [inline pkg/api/validation/genericvalidation/objectmeta.go]: impersonation is aware of service accounts and calls this indirectly from within k8s.io/apiserver.

**Pull request merged**